dia-batching-server/src/api/custom.rs

```diff
@@ -0,0 +1,138 @@
+use std::string::ToString;
+
+use crate::api::error::CustomError;
+use crate::api::Quotation;
+use crate::AssetSpecifier;
+use chrono::prelude::*;
+use graphql_client::{GraphQLQuery, Response};
+use rust_decimal::Decimal;
+
+pub struct CustomPriceApi;
+
+impl CustomPriceApi {
+	pub async fn get_price(asset: &AssetSpecifier) -> Result<Quotation, CustomError> {
+		if !Self::is_supported(asset) {
+			return Err(CustomError(format!("Unsupported asset: {:?}", asset)));
+		}
+
+		// Go through all the custom APIs and check if the asset is supported by any of them
+		if AmpePriceView::supports(asset) {
```

Comment:
It seems redundant but the first check in l. 14 is supposed to check whether the `asset` is generally supported by the `CustomPriceApi` and once we know that it's supported we would go through all the custom APIs to see which one in particular supports that asset. Since we only have the AMPE price as a custom API, it's redundant but this way it's easy to extend with more assets in the future. 

I can also remove the first check and make it iterate over all available custom APIs. Then we can get rid of the definition of `custom_assets` in l. 27. What do you think?

Reply:
I refactored it now to make the checks less redundant.

Reply:
Looks great!